service/gcs/bridge/bridge_handler_test.go

```diff
@@ -0,0 +1,800 @@
+	respWriteCount int
+}
+
+func (w *testResponseWriter) Header() *prot.MessageHeader {
+	return w.header
+}
+
+func (w *testResponseWriter) Write(r interface{}) {
+	w.response = r
+	w.respWriteCount++
+}
+
+func (w *testResponseWriter) Error(err error) {
+	w.err = err
+	w.respWriteCount++
+}
+
+func createRequest(t *testing.T, id prot.MessageIdentifier, message interface{}) *Request {
+	r := &Request{}
+
+	bytes := make([]byte, 0)
+	if message != nil {
+		var err error
+		bytes, err = json.Marshal(message)
+		if err != nil {
+			t.Fatalf("failed to marshal message for request: (%s)", err)
+		}
+	}
+	hdr := &prot.MessageHeader{
+		Type: id,
+		Size: uint32(prot.MessageHeaderSize + len(bytes)),
+		ID:   0,
+	}
+
+	r.Header = hdr
+	r.Message = bytes
+	return r
+}
+
+func createResponseWriter(r *Request) *testResponseWriter {
+	hdr := &prot.MessageHeader{
+		Type: prot.GetResponseIdentifier(r.Header.Type),
+		ID:   r.Header.ID,
+	}
+
+	return &testResponseWriter{header: hdr}
+}
+
+func verifyResponseWriteCount(t *testing.T, rw *testResponseWriter) {
+	if rw.respWriteCount != 1 {
+		t.Fatalf("response was written (%d) times != 1", rw.respWriteCount)
+	}
+}
+
+func verifyResponseError(t *testing.T, rw *testResponseWriter) {
+	verifyResponseWriteCount(t, rw)
+	if rw.err == nil {
+		t.Fatal("response did not write an error")
+	}
+}
+
+func verifyResponseJSONError(t *testing.T, rw *testResponseWriter) {
+	verifyResponseError(t, rw)
+	if !strings.Contains(rw.err.Error(), "failed to unmarshal JSON") {
+		t.Fatal("response error was not a json marshal error")
+	}
+}
+
+func verifyResponseSuccess(t *testing.T, rw *testResponseWriter) {
+	verifyResponseWriteCount(t, rw)
+	if rw.response == nil {
+		t.Fatal("response was a success but no message was included")
+	}
+}
+
+func Test_CreateContainer_InvalidJson_Failure(t *testing.T) {
+	req := createRequest(t, prot.ComputeSystemCreateV1, nil)
+	rw := createResponseWriter(req)
+
+	tb := new(Bridge)
+	tb.createContainer(rw, req)
+
+	verifyResponseJSONError(t, rw)
+}
+
+func Test_CreateContainer_InvalidHostedJson_Failure(t *testing.T) {
+	r := &prot.ContainerCreate{
+		MessageBase: &prot.MessageBase{
+			ContainerID: "test",
+			ActivityID:  "act",
+		},
+	}
+
+	req := createRequest(t, prot.ComputeSystemCreateV1, r)
+	rw := createResponseWriter(req)
+
+	tb := new(Bridge)
+	tb.createContainer(rw, req)
+
+	verifyResponseJSONError(t, rw)
+}
+
+func Test_CreateContainer_CoreCreateContainerFails_Failure(t *testing.T) {
+	r := &prot.ContainerCreate{
+		MessageBase: &prot.MessageBase{
+			ContainerID: "test",
+			ActivityID:  "act",
+		},
+		ContainerConfig: "{}", // Just unmarshal to defaults
+	}
+
+	req := createRequest(t, prot.ComputeSystemCreateV1, r)
+	rw := createResponseWriter(req)
+
+	tb := &Bridge{
+		coreint: &mockcore.MockCore{
+			Behavior: mockcore.Error,
+		},
+	}
+	tb.createContainer(rw, req)
+
+	verifyResponseError(t, rw)
+}
+
+func Test_CreateContainer_Success_WaitContainer_Failure(t *testing.T) {
```

Comment:
Isn't this a success test?

Reply:
This name is to imply the CreateContainer succeeds but the WaitContainer notification fails.

Reply:
I might be missing something, but where in the test does it test for failure?

Reply:
Does the comment you added describe how the test is testing for failure? I'm not sure I understand exactly what it's saying, and just want to make sure.